ql/src/java/org/apache/hadoop/hive/ql/processors/ShowProcesslistProcessor.java

```diff
@@ -0,0 +1,110 @@
+
+/**
+ * List operations/queries being performed in sessions within hiveserver2
+ */
+public class ShowProcesslistProcessor implements CommandProcessor{
+  private static final Logger LOG = LoggerFactory.getLogger(ShowProcesslistProcessor.class.getName());
+  private static final SessionState.LogHelper console = new SessionState.LogHelper(LOG);
+  private List<ProcessListInfo> liveQueries = null;
+
+  public void setup(List<ProcessListInfo> liveQueries) {
+    this.liveQueries = liveQueries;
+  }
+
+  /**
+   * Creates a Schema object with running operation details
+   * @return
+   */
+  private Schema getSchema(){
+    Schema sch = new Schema();
+    sch.addToFieldSchemas(new FieldSchema("User Name", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Ip Addr", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Execution Engine", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Id", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Active Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Idle Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Query ID", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("State", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Opened Timestamp", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Elapsed Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Runtime (s)", "string", ""));
+    sch.putToProperties(SERIALIZATION_NULL_FORMAT, defaultNullString);
+    return sch;
+  }
+
+  @Override
+  public CommandProcessorResponse run(String command) throws CommandProcessorException {
+    try {
+      String[] tokens = command.split("\\s+");
+      boolean isCorrectSubCommand = "processlist".equalsIgnoreCase(tokens[0].toLowerCase());
+
+      if (tokens.length != 1 || ! isCorrectSubCommand) {
+        throw new CommandProcessorException("ShowProcessList Processor Failed: Unsupported sub-command option");
+      }
+      // TODO : Authorization?
+      if (liveQueries == null || liveQueries.isEmpty()) {
+        return new CommandProcessorResponse(getSchema(),"No queries running currently");
+      }
+      SessionState ss = SessionState.get();
+      liveQueries.forEach (query -> {
+            ss.out.println(
+                Joiner.on("\t").join(
+                    query.getUserName(),
+                    query.getIpAddr(),
+                    query.getExecutionEngine(),
+                    query.getSessionId(),
+                    query.getSessionActiveTime(),
+                    query.getSessionIdleTime(),
+                    query.getQueryId(),
+                    query.getState(),
+                    query.getBeginTime(),
+                    query.getElapsedTime(),
+                    query.getRuntime()
+                ));
+          }
+      );
+      return new CommandProcessorResponse(getSchema(), null);
```

Comment:
how does the client get the result since the message is `null`?

Reply:
Results are displayed through SessionState _PrintStream_ in all these ql processors.